Software-Engineering/L5-Principal-Software-Engineer.md

```diff
@@ -1,97 +1,124 @@
+- I contributed to building blocks or core technologies used by a variety of teams.
 
 </details>
 
 ## 🌱 Teaching and Mentoring
 
 - I am an expert mentor and teacher.
+- I am curious, listen, and encourage healthy debate.
+- I allow myself to be persuaded by great ideas.
 - I create clarity by communicating complex things in simple and accessible ways.
 - I understand my broadest impact comes from empowering those around me.
 - I am an expert at designing and reviewing software architecture and code.
-- I help teams and individuals understand our technical strategy, and the things we are doing to progress it.
+- I help teams and individuals understand our technical strategy, how it connects to our product strategy, and the things we are doing to progress it.
 - I regularly provide training and support for new methods, tools, and patterns.
-- I collaborate with managers to identify and find ways to fill any gaps in technical skills required in my teams.
-- I am curious, listen, and encourage debate.
-- I allow myself to be persuaded by great ideas.
+- I actively share knowledge relevant to multiple teams.
+- I collaborate with managers and lead software engineers to identify and find ways to fill any gaps in technical skills required in my teams.
 
 <details>
 <summary>Examples</summary>
 
 - I was consistently in demand for design and code review.
-- I am actively mentoring multiple Octopus engineers.
+- I became aware that I was a bottleneck and worked proactively with managers to uplift and empower other people to make the team more effective.
+- I am actively mentoring multiple Octopus engineers, including a Lead Software Engineer aspiring to become a Principal Software Engineer.
 - Engineers at Octopus seek out to work on my teams to learn from me.
 - After reviewing a code contribution from an engineer I saw could be improved, I worked directly with that engineer and their manager or team leader to uplift their skills in the identified growth area.
 - I helped a team make a course correction based on new information or ideas.
 - I provided appropriate technical freedom, and framing, that allowed the Lead Software Engineers on my teams to thrive independently.
+- I have elevated/shifted my focus significantly because the team(s) I support have been uplifted and empowered to the point they can deliver their mission with me as a sounding board.
 
 </details>
 
 ## 🧭 Culture and Leadership
 
-- I create an environment where others feel safe, by modeling vulnerability, thinking in the open, being wrong, and accepting corrections.
-- I generate energy within the teams I work with, motivating them to achieve great outcomes.
+- I am a proponent of [operational excellence](https://en.wikipedia.org/wiki/Operational_excellence) and [learning from incidents](https://www.learningfromincidents.io/about).
```

Comment:
💭 Clarifying: Do we mean [operational excellence](https://en.wikipedia.org/wiki/Operational_excellence) in the broad sense of being excellent at operations... or in some more specific sense (e.g. [of the Juran model, Shingo model, FLEX](https://en.wikipedia.org/wiki/Operational_excellence), etc)? 
- If meant in the broad sense, suggest replacing the jargon with what we actually mean and want to highlight here
- If meant in the specific sense... I better go read that book! 😅

Reply:
My intent is they are proponents of the principles that fall under the term "operational excellence". I used the Wikipedia entry as a strawman... but now realise it's very broad.

I was originally inspired by Amazon's well-architected framework, where one pillar is called "Operational Excellence": https://docs.aws.amazon.com/wellarchitected/latest/operational-excellence-pillar/operational-excellence.html

Perhaps we could point at that? Or adapt theirs to write our own? 🤔

Reply:
It may help to break up this problem into 2 steps:

1. For this PR, simply say what we mean. That is, what is it about Amazon's "Operational Excellence" that we want to capture here, that isn't already captured elsewhere (in this role description, or in other resources like our Technical Architecture)?

2. More broadly, we could then have an initiative to evaluate and adopt some framework as a new sensible default. Then we'd raise a separate PR to point to that.

Reply:
I agree with Roy that I'd rather just say what we mean here, and resist linking to external documentation which we don't have control over. 

The Amazon link you mentioned Mike says "At Amazon, we define operational excellence as a commitment to build software correctly while consistently delivering a great customer experience" - can we use that instead? And break the incidents learning part into a separate dot point without the link?

Reply:
+1 to break out the incident learning part into a separate dot point.

Saying what we mean here without an external link would help us continue to refine this role definition with the boundary of Octopus.

Reply:
I'm going to delete this for now so we can circulate this iteration.

Reply:
Actually, I think I found a way to express it. Let me know what you think

Reply:
Just opening this back up because I'm a bit confused by this one as well. 

I think putting this in would probably trigger go away and do a little bit of learning to make sure I understand this well and even then on the face of it feels like I'd be championing a specific change or uplift rather than it just being part of the role.

But the example doesn't make me think this:
```
- During an incident postmortem, I recognised a valuable lesson and found a way to disseminate the lesson throughout R&D.
```

If I were to try to reverse engineer something from that example I might end up with:
```
- I create an environment of safety and learning with respect to incident management
```

Does that miss anything?

Reply:
Thanks all so far. What I think I'm trying to express is the importance of our most influential software engineers championing "safety and learning" across the spectrum "from discovery to delivery and incidents".

I have removed the added line (originally operation excellence and learning from incidents) and added a statement explaining that "safety and learning" is expected across the entire spectrum/lifecycle.

If the extra qualification isn't resonating ("I still don't get it"), or is redundant ("The spectrum is already clear enough without further qualification"), I'm happy to remove it.